framework: remove CMAKE_RUNTIME_OUTPUT_DIRECTORY override

arnout opened the pull request:
The CMakeLists.txt overrides the default output directory for binaries
(and for libraries as well). However, this breaks ctest, because ctest
looks for the binaries in standard locations relative to the build
directory of each test, and this common location is not one of them.

Probably, this non-standard location is a leftover from the time that
the libraries and binaries were not yet installed to the build/install
directory (as witnessed by the comment "Update output directories for
easy deployment"). Currently, however, this is no longer relevant,
thanks to proper installation to the build/install directory (or, when
using the UGW/RDKB build systems, directly into the proper location in
the rootfs).

Remove the override of `CMAKE_RUNTIME_OUTPUT_DIRECTORY`.

Ideally, the definitions of `CMAKE_LIBRARY_OUTPUT_DIRECTORY` and
`CMAKE_MULTIAP_OUTPUT_DIRECTORY` should be removed as well. However, that
is less trivial because they are also used in other places. We can keep
them as they are for the time being.

tomereli commented:
This was done for easier deploy, we do the same in the controller agent and common so that in deploy we only look at build/<module>/out

Using install duir has other modules so it will be more complex. Is there another way to accomplish that?

Commit:
framework: build: fix ctest not finding tests

CMakeLists.txt overrides the default output directory
CMAKE_RUNTIME_OUTPUT_DIRECTORY (This is done in order to make deploying
easier, e.g. all outputs are found by deploy script in
build/<module>/out instead of having a mapping of what to deploy from
build/install where all modules outputs are installed to.

This causes CTEST not to find the tests, so fix that by explicitely
setting the tests binaries location by using
`add_test(NAME test COMMAND $<TARGET_FILE>:test>)` format.

Signed-off-by: Tomer Eliyahu <tomer.b.eliyahu@intel.com>

tomereli commented:
Ah, so many git mistakes in one simple change - @arnout I'm sorry, I didn't mean to force push your branch - by coincidence, we actually chose the same branch name for the fix. Anyway, I was meaning to ask you to review this fix instead of your suggested fix which will break the deploy script for none native archs.
